src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -311,22 +316,34 @@ internal Expr BindArrayIndexCore(Expr pOp1, Expr pOp2)
             Expr transformedIndices = pOp2.Map(
                 x =>
                 {
-                    Expr pTemp = binder.mustConvert(x, pDestType);
+                    Expr pTemp = binder.MustConvertWithSuppressedMessage(x, pDestType);
                     return pDestType == pIntType
                         ? pTemp
-                        : ExprFactory.CreateCast(EXPRFLAG.EXF_INDEXEXPR, pDestType, pTemp);
+                        : ExprFactoryCreateCastWithSuppressedMessage(EXPRFLAG.EXF_INDEXEXPR, pDestType, pTemp);
                 });
 
             // Allocate a new expression, the type is the element type of the array.
             // Array index operations are always lvalues.
             return ExprFactory.CreateArrayIndex(elementType, pOp1, transformedIndices);
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
+            Justification = "Workarounds mono/linker#1416. All usages are marked as unsafe.")]
```

Comment:
(nit) this is totally my personal preference, but I prefer to have full URLs to allow for easy clicking, and to ensure the URL is pointing to the correct issue.
```suggestion
            Justification = "Works around https://github.com/mono/linker/issues/1416. All usages are marked as unsafe.")]
```